logstash-core-event-java/spec/event_spec.rb

```diff
@@ -90,6 +90,12 @@
       expect(e["[foo][2]"]).to eq(1.0)
       expect(e["[foo][3]"]).to be_nil
     end
+
+    it "should set nil value" do
+      e = LogStash::Event.new()
+      e["[foo]"] = nil
+      expect(e.to_hash).to include("foo")
```

Comment:
proposal, what about having the condition like ```expect(e.to_hash).to include("foo" => nil)``` ? this will check the existence of the pair in the hole, that the key exist and the value is nil.